script.js

```diff
@@ -1,4 +1,1250 @@
+    (userGuess.toLowerCase() == "spock" &&
+      (gameHand == "scissors" || gameHand == "stone")) ||
+    (userGuess.toLowerCase() == "lizard" &&
+      (gameHand == "paper" || gameHand == "spock"))
+  ) {
+    messsageOutput = userWin;
+    nmbrOfWins = nmbrOfWins + 1;
+    prevWinner = userName;
+  } else {
+    messsageOutput = errorMessage;
+  }
+
+  console.log("prevWinner=");
+  console.log(prevWinner);
+
+  var totalGamesPlayed = nmbrOfDraws + nmbrOfLosses + nmbrOfWins;
+
+  var scoreDisplay = `<br><br> Wins - Losses <br> *** ${nmbrOfWins} - ${nmbrOfLosses} ***<br>Number of ties(draw) = ${nmbrOfDraws}<br>Number of Games Played = ${totalGamesPlayed}<br> Win Percentage =${percentage(
+    nmbrOfWins,
+    totalGamesPlayed
+  ).toFixed(1)}% . `;
+
+  var prevWinnerDisplay = `<br>Winner of Most Recent Round = ${prevWinner}<br>`;
+
+  if (messsageOutput == errorMessage) {
+    scoreDisplay = "";
+    prevWinnerDisplay = "";
+    endingMessage = "";
+  } else if (messsageOutput == normalModeWarning) {
+    modeTitle = "";
+    scoreDisplay = "";
+    prevWinnerDisplay = "";
+    endingMessage = "";
+  } else if (messsageOutput == tryAgain) {
+    scoreDisplay = "";
+    endingMessage = "";
+  }
+
+  return (
+    modeTitle +
+    messsageOutput +
+    scoreDisplay +
+    prevWinnerDisplay +
+    endingMessage
+  );
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var welcomeMsg = `, welcome to the Normal Mode of Scissors, Paper, Stone. <br><br> Scissors > Paper > Stone and Stone > Scissors. 
+                      <br><br> You get 1 point for every round won, no points are awarded for draws. 
+                      <br><br> Type '<i>korean</i>' to enable MUK-JJI-PPA mode.
+                          <br> <i>only activate MUK-JJI-PPA mode when there is a winner in the round.</i>      
+                      <br><br> Type '<i>StarTrek</i> to enable Scissors Paper Stone Spock Lizard mode.
+                      <br><br> Please submit input "normal" in any non-Normal modes to before switching to other variant game modes
+                      `;
+  if (currentMode == "waiting for username") {
+    userName = input;
+    currentMode = "NormalSPS";
+    myOutputValue = "Hello " + userName + welcomeMsg;
+  } else if (currentMode == "NormalSPS" && input.toLowerCase() == "korean") {
+    currentMode = "korean";
+    myOutputValue = koreanModeWarning;
+  } else if (currentMode == "korean") {
+    myOutputValue = playMukJjiPpa(userName, input);
+  } else if (currentMode == "NormalSPS" && input.toLowerCase() == "startrek") {
+    currentMode = "startrek";
+    myOutputValue = spockLizardWarning;
+  } else if (currentMode == "startrek") {
+    myOutputValue = playSpockLizard(userName, input);
+  } else if (currentMode == "NormalSPS") {
+    myOutputValue = playNormalSPS(userName, input);
+  }
+  console.log("Currrent Mode=");
+  console.log(currentMode);
   return myOutputValue;
 };
```

Comment:
Good job on this! I already see the improvement from the prev commented out code to this. Nice job implementing a custom version of SPS!